packages/tpetra/core/src/Tpetra_Details_crsUtils.hpp

```diff
@@ -112,37 +112,39 @@ pad_crs_arrays(
   if (additional_size_needed == 0)
     return;
 
+  using ptrs_value_type = typename RowPtr::non_const_value_type;
+  using inds_value_type = typename Indices::non_const_value_type;
+  using vals_value_type = typename Values::non_const_value_type;
+
   // The indices array must be resized and the row_ptr arrays shuffled
   auto indices_new = uninitialized_view<Indices>("ind new", indices.size()+additional_size_needed);
   auto values_new = uninitialized_view<Values>("val new", pad_values ? values.size()+additional_size_needed : 0);
-  Kokkos::deep_copy(values_new, Teuchos::ScalarTraits<typename Values::value_type>::zero());
+  Kokkos::deep_copy(values_new, vals_value_type(0.0));
```

Comment:
You could also use `Kokkos::ArithTraits<vals_value_type>::zero()`, though this is fine (even for integer or complex types).